core/textile/model/sent_file.go

```diff
@@ -0,0 +1,177 @@
+
+	if existingFile, err := m.FindSentFile(ctx, file.DbId, file.Bucket, file.Path); err == nil {
+		log.Debug("Model.CreateSentFileViaInvitation: file already in the collection")
+		return existingFile, nil
+	}
+
+	newInstance := &SentFileSchema{
+		ID: "",
+		SentFileViaInvitationSchema: SentFileViaInvitationSchema{
+			DbID:          file.DbId,
+			Bucket:        file.Bucket,
+			Path:          file.Path,
+			InvitationId:  invitationID,
+			BucketKey:     file.BucketKey,
+			EncryptionKey: key,
+		},
+		CreatedAt: time.Now().UnixNano(),
+	}
+
+	return m.createSentFile(ctx, newInstance)
+}
+
+func (m *model) createSentFile(ctx context.Context, instance *SentFileSchema) (*SentFileSchema, error) {
+	metaCtx, metaDbID, err := m.initSentFileModel(ctx)
+	if err != nil && metaDbID == nil {
+		return nil, err
+	}
+
+	instances := client.Instances{instance}
+	res, err := m.threads.Create(metaCtx, *metaDbID, sentFileModelName, instances)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debug(fmt.Sprintf("Model.createSentFile: stored sent file res=%+v", res))
+
+	id := res[0]
+	return &SentFileSchema{
+		ID:                          core.InstanceID(id),
+		SentFileViaInvitationSchema: instance.SentFileViaInvitationSchema,
+		CreatedAt:                   instance.CreatedAt,
+	}, nil
+}
+
+// Finds the metadata of a file that has been shared by the user
+func (m *model) FindSentFile(ctx context.Context, remoteDbID, bucket, path string) (*SentFileSchema, error) {
+	metaCtx, dbID, err := m.initSentFileModel(ctx)
+	if err != nil || dbID == nil {
+		return nil, err
+	}
+
+	rawFiles, err := m.threads.Find(metaCtx, *dbID, sentFileModelName, db.Where("dbId").Eq(remoteDbID).And("bucket").Eq(bucket).And("path").Eq(path), &SentFileSchema{})
+	if err != nil {
+		return nil, err
+	}
+
+	if rawFiles == nil {
+		return nil, errSentFileNotFound
+	}
+
+	files := rawFiles.([]*SentFileSchema)
+	if len(files) == 0 {
+		return nil, errSentFileNotFound
+	}
+
+	log.Debug(fmt.Sprintf("Model.FindSentFile: returning files=%+v", files))
+
+	return files[0], nil
+}
+
+// Lists the metadata of files sent by the user
+// If seek == "", will start looking from the beginning. If it's an existing ID it will start looking from that ID.
+func (m *model) ListSentFiles(ctx context.Context, seek string, limit int) ([]*SentFileSchema, error) {
+	metaCtx, dbID, err := m.initSentFileModel(ctx)
+	if err != nil || dbID == nil {
+		return nil, err
+	}
+
+	query := db.OrderBy("CreatedAt").LimitTo(limit)
+
+	if seek != "" {
+		query = query.SeekID(core.InstanceID(seek))
+	}
+
+	rawFiles, err := m.threads.Find(metaCtx, *dbID, sentFileModelName, query, &SentFileSchema{})
+	if err != nil {
+		return nil, err
+	}
+
+	if rawFiles == nil {
+		return []*SentFileSchema{}, nil
+	}
+
+	files := rawFiles.([]*SentFileSchema)
+	return files, nil
+}
+
+// XXX: this is to reuse the builders in the sharing.go
+func (sf *SentFileSchema) ReceivedFileSchema() *ReceivedFileSchema {
```

Comment:
Please call this in `GetAllCollectionConfigs` in model.go so that it gets restored too.

Reply:
@dmerrill6 

Thank you for spotting this!